indexing-service/src/main/java/io/druid/indexing/common/task/HadoopConverterTask.java

```diff
@@ -0,0 +1,408 @@
+   * @param hadoopDependencyCoordinates List of hadoop coordinates to pull in
+   * @param distributedSuccessCache     The URI to use as the working path for hadoop
+   * @param jobPriority                 The priority of the hadoop job. Will be unset if left null
+   * @param segmentOutputPath           The output path to put the segment
+   *
+   * @return A SegmentConverterTask for the datasource's interval with the indexSpec specified.
+   */
+  public static HadoopConverterTask create(
+      String dataSource,
+      Interval interval,
+      IndexSpec indexSpec,
+      boolean force,
+      boolean validate,
+      List<String> hadoopDependencyCoordinates,
+      URI distributedSuccessCache,
+      String jobPriority,
+      String segmentOutputPath,
+      String classpathPrefix
+  )
+  {
+    final String id = makeId(dataSource, interval);
+    return new HadoopConverterTask(
+        id,
+        id,
+        dataSource,
+        interval,
+        null,
+        indexSpec,
+        force,
+        validate,
+        hadoopDependencyCoordinates,
+        distributedSuccessCache,
+        jobPriority,
+        segmentOutputPath,
+        classpathPrefix
+    );
+  }
+
+  /**
+   * Create a task to update the segment specified to the most recent binary version with the specified indexSpec
+   *
+   * @param segment   The segment to which this update should be applied
+   * @param indexSpec The IndexSpec to use in the updated segments
+   * @param force     Force an update, even if the task thinks it doesn't need to update.
+   * @param validate  Validate the new segment compared to the old segment on a row by row basis
+   *
+   * @return A SegmentConverterTask for the segment with the indexSpec specified.
+   */
+  public static HadoopConverterTask create(
+      DataSegment segment,
+      IndexSpec indexSpec,
+      boolean force,
+      boolean validate,
+      List<String> hadoopDependencyCoordinates,
+      URI distributedSuccessCache,
+      String jobPriority,
+      String segmentOutputPath,
+      String classpathPrefix
+  )
+  {
+    final Interval interval = segment.getInterval();
+    final String dataSource = segment.getDataSource();
+    final String id = makeId(dataSource, interval);
+    return new HadoopConverterTask(
+        id,
+        id,
+        dataSource,
+        interval,
+        segment,
+        indexSpec,
+        force,
+        validate,
+        hadoopDependencyCoordinates,
+        distributedSuccessCache,
+        jobPriority,
+        segmentOutputPath,
+        classpathPrefix
+    );
+  }
+
+  private static String makeId(String dataSource, Interval interval)
+  {
+    Preconditions.checkNotNull(dataSource, "dataSource");
+    Preconditions.checkNotNull(interval, "interval");
+    return joinId(TYPE, dataSource, interval.getStart(), interval.getEnd(), new DateTime());
+  }
+
+  protected HadoopConverterTask(
```

Comment:
This is a legacy from the ConvertTask but it also works around the oddities of having null segments which the original convert task supported. I'll give it a once over and see if I can simplify it.